app/frontend/wallet/shelley-wallet.ts

```diff
@@ -260,14 +286,24 @@ const ShelleyWallet = ({
     }
   }
 
-  function prepareTxAux(plan) {
+  async function calculateTtl() {
```

Comment:
As mentioned, I would get the time of the beginning of the Shelley era and add the time passed since then.

Reply:
Fixed